typings/algoliasearch.d.ts

```diff
@@ -0,0 +1,1252 @@
+// Type definitions for algoliasearch-client-js 3.18.1
+// Project: http://github.com/algolia/algoliasearch-client-js
+
+/*
+ Interface for the algolia client object
+ */
+interface IAlgoliaClient {
+    /**
+     * Initialization of the client
+     * @param name: index name
+     * return algolia index object
+     * https://github.com/algolia/algoliasearch-client-js#init-index---initindex
+     */
+    initIndex(name:string):IAlgoliaIndex;
+    /**
+     * Initialization of the client
+     * @param name of the index
+     * @param params options related to the client initialization
+     * return algolia index object
+     * https://github.com/algolia/algoliasearch-client-js#init-index---initindex
+     */
+    initIndex(name:string, params:IClientOptions):IAlgoliaIndex;
+    /**
+     * Query on multiple index
+     * @param queries index name, query and query parameters
+     * @param cb callback(err, res)
+     * https://github.com/algolia/algoliasearch-client-js#multiple-queries---multiplequeries
+     */
+    search(queries:{ indexName:string, query:string, options:IAlgoliaQueryParameters }, cb:Function);
+    /**
+     * Query on multiple index
+     * @param queries index name, query and query parameters
+     * @param cb callback(err, res)
+     * https://github.com/algolia/algoliasearch-client-js#multiple-queries---multiplequeries
+     */
+    search(queries:{ indexName:string, query:string, options:IAlgoliaQueryParameters }, params:any, cb:Function);
+    /**
+     * Query on multiple index
+     * @param queries index name, query and query parameters
+     * @param params query parameter
+     * @param cb callback(err, res)
```

Comment:
Can we add that every (asynchronous) method returns a Promise if not callback passed?